internal/api/export/worker.go

```diff
@@ -145,8 +151,49 @@ func (s *Server) exportBatch(ctx context.Context, eb *model.ExportBatch) error {
 		objectNames = append(objectNames, objectName)
 	}
 
-	// Create the index file. The index file includes _all_ batches for an ExportConfig, so multiple
-	// workers may be racing to update it. We use a lock to make them line up after one another.
+	// Emit the index file if needed.
+	if batchSize > 0 || emitIndexForEmptyBatch {
```

Comment:
This is the essence of this PR. The rest of the large diff is due to a function extraction.